src/format/KeePass2Repair.h

```diff
@@ -37,13 +38,12 @@ class KeePass2Repair
         RepairFailed
     };
 
-    KeePass2Repair();
     RepairResult repairDatabase(QIODevice* device, const CompositeKey& key);
-    Database* database() const;
+    Database* takeDatabase();
     QString errorString() const;
 
 private:
-    Database* m_db;
+    QScopedPointer<Database> m_db{nullptr};
```

Comment:
You should initialize members in the initialization list of the constructor.

Reply:
Why do you think so? User defined constructors for member default initialization only are considered not-the-best practice: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Rc-in-class-initializer

Reply:
For consistency reasons. But if you must initialize things like this, at least use the assignment syntax.
In this case, however, you could (and should)  skip initialization completely, which will default-initialize the smart pointer.

Reply:
Agree, `QScopedPointer` is initialized to `Q_NULLPTR` by default. Fixed